Lib/compileall.py

```diff
@@ -254,9 +254,9 @@ def compile_file(fullname, ddir=None, force=False, rx=None, quiet=0,
                 else:
                     print('*** ', end='')
                 # escape non-printable characters in msg
-                msg = err.msg.encode(sys.stdout.encoding,
-                                     errors='backslashreplace')
-                msg = msg.decode(sys.stdout.encoding)
+                encoding = sys.stdout.encoding if sys.stdout.encoding \
+                    else sys.getdefaultencoding()
```

Comment:
That could be rewritten to:
```suggestion
                encoding = sys.stdout.encoding or sys.getdefaultencoding()
```